src/Sidekick.Core/Initialization/Initializer.cs

```diff
@@ -9,46 +10,28 @@ namespace Sidekick.Core.Initialization
 {
     public class Initializer : IInitializer
     {
-        private readonly IServiceProvider serviceProvider;
-        private readonly ILogger logger;
-
-        public Initializer(IServiceProvider serviceProvider,
-            ILogger logger)
-        {
-            this.serviceProvider = serviceProvider;
-            this.logger = logger;
+        public bool IsReady { get; private set; }
 
-            resetServices = typeof(IOnReset).GetImplementedInterface();
-            beforeInitServices = typeof(IOnBeforeInit).GetImplementedInterface();
-            initServices = typeof(IOnInit).GetImplementedInterface();
-            afterInitServices = typeof(IOnAfterInit).GetImplementedInterface();
-        }
+        private List<IOnReset> resetServices;
+        private List<IOnBeforeInit> beforeInitServices;
+        private List<IOnInit> initServices;
+        private List<IOnAfterInit> afterInitServices;
 
-        private List<Type> resetServices { get; set; }
-        private List<Type> beforeInitServices { get; set; }
-        private List<Type> initServices { get; set; }
-        private List<Type> afterInitServices { get; set; }
-        public bool IsReady { get; private set; }
+        private readonly IServiceProvider serviceProvider;
 
-        private List<Type> GetServiceInterfaces(List<Type> implementations)
+        public Initializer(IServiceProvider serviceProvider)
```

Comment:
Can we ask the DI for the List (or IEnumerable) directly in the constructor instead of getting the services later in the process? I think you showed that in our previous conversation. If it is possible, I would get the implementations in the constructor.

Reply:
Normally it would work. Right nowbecause some of the implementations have the initializer as a dependency, trying to

Reply:
Ok Yeah, thanks. :)